### PR TITLE
Drop-down widgets for extension functions via UnresolvedSymbol

### LINKED ISSUE #6955: @widgets for extension functions of `Table` may not be found
As experienced in #6925, the `primary_key` annotation on `select_into_database_table` operation seems to be not registered.

### Actual behaviour

image

The widget on `primary_key` is a Vector Editor but it only allows to add a `Nothing`, not the actual column names as expected. Checking `Widgets.get_widget_json` we can see that the annotation for `@primary_key` seems to not be applied for some reason.

### Expected behaviour

The `"primary_key"` widget JSON should not be `null` - it is defined to be `Widget_Helpers.make_column_name_vector_selector`.

The above should then allow the vector editor to display the proper column selector dropdown.

### Context

We checked if this may be due to the fact that `select_into_database_table` was defined on renamed types:
```
import Standard.Table.Data.Table as In_Memory_Table

...

@primary_key ...
In_Memory_Table.select_into_database_table : ...
In_Memory_Table.select_into_database_table self ... primary_key ... =
    ...
```
but we split the extensions file into separate versions for in-memory and Database Tables, allowing to get rid of the renames and the issue persisted, so it is probably something else.

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Fixes #6955 by:
- using `visualisationModule` to specify the module where the visualization is to be used
- referring to method in `Meta.get_annotation` with `.method_name` - e.g. unresolved symbol notation
- evaluating arguments to `Meta.get_annotation` in the context of the user module (which can access the extension functions)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
`primary_key` is really drop down widget seems to be working:

obrazek

**Comment by JaroslavTulach** [comment]
[Test failure](https://github.com/enso-org/enso/actions/runs/5357026507/jobs/9717312587?pr=7115#step:10:5552) improved in 673c49e

File: app/gui/controller/engine-protocol/src/language_server/tests.rs [hunk 1]
```diff
@@ -479,6 +480,7 @@ fn test_execution_context() {
     };
     let positional_arguments_expressions = vec!["foo"].iter().map(|x| x.to_string()).collect();
     let visualisation_config = VisualisationConfiguration {
+        visualisation_module: "local.Widgets.Main",
```
Comment (radeusgd) [inline app/gui/controller/engine-protocol/src/language_server/tests.rs]: This string is used in so many places, it feels like it should be a constant.

btw. what does it point to? I'm not aware of a library under this path.

File: test/Tests/src/Semantic/Meta_Spec.enso [hunk 2]
```diff
@@ -280,17 +280,17 @@ spec =
 
         Test.specify "get annotations" <|
             Meta.get_annotation Meta_Spec "test_method" "a" . should_equal 7
-            Meta.get_annotation Meta_Spec "test_method" "b" . should_equal (Test_Type.Value 49)
+            Meta.get_annotation Meta_Spec .test_method "b" . should_equal (Test_Type.Value 49)
```
Comment (radeusgd) [inline test/Tests/src/Semantic/Meta_Spec.enso]: This looks so much better

File: engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/GetAnnotationNode.java [hunk 3]
```diff
@@ -39,16 +38,21 @@ Object doExecute(
...
-      Function methodFunction = scope.lookupMethodDefinition(targetType, methodName);
+      Function methodFunction;
+      String methodName = null;
+      if (method instanceof UnresolvedSymbol symbol) {
+        methodFunction = symbol.resolveFor(targetType);
+      } else {
+        methodName = expectStringNode.execute(method);
+        ModuleScope scope = targetType.getDefinitionScope();
+        methodFunction = scope.lookupMethodDefinition(targetType, methodName);
+       }
```
Comment (radeusgd) [inline engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/GetAnnotationNode.java]: Why do we still allow 'raw text' input here?

It seems that the API would be simpler if we had one input type - `UnresolvedSymbol`.

File: app/gui/controller/engine-protocol/src/language_server/tests.rs (hunk 1 above)
Comment (JaroslavTulach) [reply app/gui/controller/engine-protocol/src/language_server/tests.rs]: The string points to a file on my local disk. The string may be used few times, but only in tests, not in production code. The `visualisation_module` argument needs to be set to something. It (very likely) doesn't matter what the value is in the existing tests.

File: engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/GetAnnotationNode.java (hunk 3 above)
Comment (JaroslavTulach) [reply engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/GetAnnotationNode.java]: I am used to keep compatibility at almost all cost. The cost of keeping previous behavior in this case is low, acceptable. The idea of dropping support for the `Text` argument hasn't even come to my mind.

**Comment by JaroslavTulach** [comment]
There are two failing tests:
- [org.enso.languageserver.websocket.json.VisualisationOperationsTest](https://github.com/enso-org/enso/actions/runs/5357596016/jobs/9718673659?pr=7115)
- [RuntimeVisualisationsTest](https://github.com/enso-org/enso/actions/runs/5357596015/jobs/9718673771?pr=7115#step:10:5921)

File: app/gui/controller/engine-protocol/src/language_server/tests.rs (hunk 1 above)
Comment (JaroslavTulach) [reply app/gui/controller/engine-protocol/src/language_server/tests.rs]: Correction: sometimes it does matter, like in 09254d3

File: engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/GetAnnotationNode.java (hunk 3 above)
Comment (radeusgd) [reply engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/GetAnnotationNode.java]: The cost is that we increase the complexity of this function - making it harder to understand, test  and maintain if we need to change it in the future.

It's an internal function so I'm not sure what is the benefit of that? Especially as the old behaviour is actually leading to wrongly resolved methods - so we should discourage using it. If we still allow to use it, someone later may forget why this change was introduced, think that using the `Text` API is easier to use - and use it introducing a bug that the extension methods will not be correctly resolved somewhere.

File: app/gui/src/controller/graph/executed.rs [hunk 4]
```diff
@@ -96,7 +96,7 @@ pub struct Handle {
     execution_ctx: model::ExecutionContext,
     /// The handle to project controller is necessary, as entering nodes might need to switch
     /// modules, and only the project can provide their controllers.
-    project:       model::Project,
+    pub project:   model::Project,
```
Comment (farmaazon) [inline app/gui/src/controller/graph/executed.rs]: Why do we need this?

File: app/gui/controller/engine-protocol/src/language_server/types.rs [hunk 5]
```diff
@@ -713,6 +713,8 @@ pub type ExpressionId = Uuid;
 #[serde(rename_all = "camelCase")]
 #[allow(missing_docs)]
 pub struct VisualisationConfiguration {
+    /// FQN of module to use the visualisation from
+    pub visualisation_module: String,
```
Comment (farmaazon) [inline app/gui/controller/engine-protocol/src/language_server/types.rs]: I don't quite understand these docs. We do not take visualizations from module, but from `visualization` directory (or use a built-in one).

I think the definition in `execution_context::Visualization` is better, it could be just repeated here.

File: app/gui/src/model/execution_context.rs [hunk 6]
```diff
@@ -303,6 +303,8 @@ pub struct Visualization {
     pub id:             VisualizationId,
     /// Expression that is to be visualized.
     pub expression_id:  ExpressionId,
+    /// module to evaluate visualization in context of
+    pub module:         String,
```
Comment (farmaazon) [inline app/gui/src/model/execution_context.rs]:
```suggestion
    /// Module to evaluate visualization in context of.
    pub module:         String,
```

File: app/gui/src/controller/searcher.rs [hunk 7]
```diff
@@ -574,7 +574,8 @@ impl Searcher {
             "Standard.Visualization.AI",
             "build_ai_prompt",
         )?;
-        let vis = Visualization::new(this.id, vis_ptr, vec![]);
+        let vis =
+            Visualization::new("Standard.Visualization.AI".to_string(), this.id, vis_ptr, vec![]);
```
Comment (farmaazon) [inline app/gui/src/controller/searcher.rs]: `to_owned` is always better than `to_string` for any `&str`. Shortly: because `to_string` may make unnecessary allocations - see https://users.rust-lang.org/t/to-string-vs-to-owned-for-string-literals/1441 for details.

This comment applies to other `to_strings` as well.

File: app/gui/src/controller/visualization/manager.rs [hunk 8]
```diff
@@ -285,11 +286,15 @@ impl Manager {
             // Early return: requested to remove visualization that was already removed.
             return;
         };
+        let prj = self.executed_graph.module_qualified_name(&*self.executed_graph.project);
+        let graph = self.executed_graph.graph();
+        let module = prj.to_string() + "." + graph.module.name();
```
Comment (farmaazon) [inline app/gui/src/controller/visualization/manager.rs]:
```suggestion
        let module = format!("{prj}.{}", graph.module.name());
```

**Review by farmaazon — APPROVED** [review]
Checked the rust part, left some comments. @Frizi please also do the review here, as I don't know how it affects our widgets.

File: app/gui/src/controller/visualization/manager.rs [hunk 9]
```diff
@@ -285,11 +286,15 @@ impl Manager {
             // Early return: requested to remove visualization that was already removed.
             return;
         };
+        let prj = self.executed_graph.module_qualified_name(&*self.executed_graph.project);
```
Comment (JaroslavTulach) [inline app/gui/src/controller/visualization/manager.rs]: This is the line that requires [project to be public](https://github.com/enso-org/enso/pull/7115#discussion_r1242031934). Suggestions to change this code to not require `pub project` are welcomed.

File: app/gui/src/controller/graph/executed.rs (hunk 4 above)
Comment (JaroslavTulach) [reply app/gui/src/controller/graph/executed.rs]: Without making the `project` field `pub` the [following line](https://github.com/enso-org/enso/pull/7115/files?file-filters%5B%5D=.rs&show-viewed-files=true#r1242077660) doesn't compile.

File: app/gui/src/controller/searcher.rs (hunk 7 above)
Comment (Frizi) [reply app/gui/src/controller/searcher.rs]: This recommendation is from 2015 and is since outdated. Later in the same thread:

> Now that [specialization for str::to_string() 996](https://github.com/rust-lang/rust/pull/32586) has landed, we can safely say that to_string() has the same performance as to_owned()

Now we might have other code style reasons to prefer one over the other, but performance is not one of them.

Comment (JaroslavTulach) [reply app/gui/src/controller/searcher.rs]: I was just about to do search and replace, but less work is obviously better. Delaying `to_owned` changes for now.

File: app/gui/src/controller/visualization/manager.rs (hunk 9 above)
Comment (farmaazon) [reply app/gui/src/controller/visualization/manager.rs]: I think executed graph's `module_qualified_name` (https://github.com/enso-org/enso/blob/3e6af201dbecffdedd3b6dffde7d51957bfcec9a/app/gui/src/controller/graph/executed.rs) should not take project as an argument, but take the project it owns instead.

File: app/gui/controller/engine-protocol/src/language_server/types.rs (hunk 5 above)
Comment (JaroslavTulach) [reply app/gui/controller/engine-protocol/src/language_server/types.rs]: Changed in 1223fbaa51

File: app/gui/src/controller/visualization/manager.rs [hunk 10]
```diff
@@ -285,11 +286,15 @@ impl Manager {
...
+        let graph = self.executed_graph.graph();
+        let module = format!("{prj}.{}", graph.module.name());
```
Comment (Frizi) [inline app/gui/src/controller/visualization/manager.rs]: I believe this should be changed to

```rs
let graph = self.executed_graph.graph();
let module = graph.qualified_name();
```
The graph now contains the `project_name`, but its `qualified_name` method doesn't make use of it yet, so it has to be adjusted as well.

```rs
    // gui/src/controller/module.rs impl Handle

    /// Get the module's qualified name.
    pub fn qualified_name(&self) -> QualifiedName {
        QualifiedName::new_module(self.project_name.clone(), self.model.id())
    }
```

File: app/gui/src/model/execution_context.rs [hunk 11]
```diff
@@ -303,6 +303,8 @@ pub struct Visualization {
     pub id:             VisualizationId,
     /// Expression that is to be visualized.
     pub expression_id:  ExpressionId,
+    /// Module to evaluate visualization in context of.
+    pub module:         String,
```
Comment (Frizi) [inline app/gui/src/model/execution_context.rs]: This should probably be a `QualifiedName`.

File: app/gui/src/controller/graph/widget.rs [hunk 12]
```diff
@@ -410,6 +410,10 @@ impl QueryData {
         Ast::raw_text_literal(arg).repr()
     }
 
+    fn as_unresolved_symbol(arg: &str) -> String {
+        ".".to_string() + arg
```
Comment (Frizi) [inline app/gui/src/controller/graph/widget.rs]: It's not clear to me what "unresolved symbol" means in this context. The syntax also doesn't make sense to me initially, so it warrants some expanation as a code comment.

Also, it would be better to use formatting here: `format!(".{arg}")`. Probably doesn't even have to be a dedicated function call, and can be directly inlined into `visualization_metadata` (with comment).

File: app/gui/src/controller/visualization/manager.rs [hunk 13]
```diff
@@ -158,6 +158,7 @@ impl Default for Status {
 #[allow(missing_docs)]
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Desired {
+    pub module:           String,
```
Comment (Frizi) [inline app/gui/src/controller/visualization/manager.rs]: Should be `QualifiedName`.

**Review by Frizi — CHANGES_REQUESTED** [review]

File: app/gui/src/model/execution_context.rs (hunk 11 above)
Comment (JaroslavTulach) [reply app/gui/src/model/execution_context.rs]: OK, done in 8f60565a0

File: app/gui/src/controller/graph/widget.rs (hunk 12 above)
Comment (JaroslavTulach) [reply app/gui/src/controller/graph/widget.rs]: Done in 3ffb6e3249

File: app/gui/src/controller/graph/executed.rs (hunk 4 above)
Comment (JaroslavTulach) [reply app/gui/src/controller/graph/executed.rs]: Gone in 429001e689

File: app/gui/src/controller/visualization/manager.rs (hunk 9 above)
Comment (JaroslavTulach) [reply app/gui/src/controller/visualization/manager.rs]: Done in [429001e](https://github.com/enso-org/enso/pull/7115/commits/429001e689408a191dd952b4e746a9caa8fd58c4)

File: app/gui/src/controller/visualization/manager.rs (hunk 13 above)
Comment (JaroslavTulach) [reply app/gui/src/controller/visualization/manager.rs]: Done in [8f60565](https://github.com/enso-org/enso/pull/7115/commits/8f60565a0b981a02ef433cbf1db9b5fb8e116225)

File: app/gui/controller/engine-protocol/src/language_server/tests.rs (hunk 1 above)
Comment (radeusgd) [reply app/gui/controller/engine-protocol/src/language_server/tests.rs]: I was analysing what `local.Widgets.Main` could be pointing to. Couldn't figure it out and was confused. Now I know that the file was just not existing in the repo.

I think it will be clearer if the name reflects this, maybe let's call it then `local.Nonexistent_Arbitrary_Project.Main`?

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala [hunk 14]
```diff
@@ -302,13 +312,19 @@ object UpsertVisualizationJob {
     * @return either the evaluation result or an evaluation error
     */
   private def evaluateVisualizationExpression(
+    module: String,
```
Comment (4e6) [inline engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:317]: Forgot to document new parameter

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala [hunk 15]
```diff
@@ -224,6 +228,7 @@ object UpsertVisualizationJob {
   private def evaluateModuleExpression(
     module: Module,
     expression: Api.VisualizationExpression,
+    expressionModule: Module,
```
Comment (4e6) [inline engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:232]: You forgot to update the doc

File: engine/polyglot-api/src/main/scala/org/enso/polyglot/runtime/Runtime.scala [hunk 16]
```diff
@@ -618,18 +618,16 @@ object Runtime {
       */
     case class VisualizationConfiguration(
       executionContextId: ContextId,
-      expression: VisualizationExpression
+      expression: VisualizationExpression,
+      visualizationModule: String
```
Comment (4e6) [inline engine/polyglot-api/src/main/scala/org/enso/polyglot/runtime/Runtime.scala:623]: Update the doc

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala [hunk 17]
```diff
@@ -302,13 +312,19 @@ object UpsertVisualizationJob {
...
   ): Either[EvaluationFailure, EvaluationResult] = {
     for {
-      module     <- findModule(expression.module)
-      expression <- evaluateModuleExpression(module, expression)
+      module           <- findModule(module)
+      expressionModule <- findModule(expression.module)
+      expression <- evaluateModuleExpression(
+        module,
+        expression,
+        expressionModule
```
Comment (4e6) [inline engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:328]: Why do we need both `module` and `expressionModule`?

File: engine/language-server/src/main/scala/org/enso/languageserver/runtime/VisualizationConfiguration.scala [hunk 18]
```diff
@@ -99,11 +103,15 @@ object VisualizationConfiguration {
             expression <- cursor
               .downField(CodecField.Expression)
               .as[MethodPointer]
+            visualizationModule <- cursor
+              .downField(CodecField.VisualizationModule)
+              .as[String]
```
Comment (4e6) [inline engine/language-server/src/main/scala/org/enso/languageserver/runtime/VisualizationConfiguration.scala:108]: You made the `visualizationModule` field required. Please update the `protocol-language-server.md` API doc.

**Review by 4e6 — CHANGES_REQUESTED** [review]

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala (hunk 17 above)
Comment (JaroslavTulach) [reply engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:328]: One is used for `prepareFunctionCall` and the other one for `evaluateExpression`.

File: app/gui/src/model/execution_context.rs (hunk 11 above)
Comment (JaroslavTulach) [reply app/gui/src/model/execution_context.rs]: Alas, @Frizi, it may not have been good idea:

> [`QualifiedName`] always omit the "Main" module name in its path

But I need to construct `local.Project.Main` module name!

**Resolved** in b0a86fbc6e.

File: app/gui/src/controller/visualization/manager.rs (hunk 10 above)
Comment (JaroslavTulach) [reply app/gui/src/controller/visualization/manager.rs]: Changed to:
```
        let prj = self.executed_graph.module_qualified_name();
        let graph = self.executed_graph.graph();
        let module = prj.new_child(graph.module.name());
```

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala (hunk 14 above)
Comment (JaroslavTulach) [reply engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:317]: Documented in 1042ac9e24

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala (hunk 15 above)
Comment (JaroslavTulach) [reply engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:232]: Documented in [1042ac9](https://github.com/enso-org/enso/pull/7115/commits/1042ac9e24f9d4c58f8f40a7f5dbbce698b4361e)

File: engine/polyglot-api/src/main/scala/org/enso/polyglot/runtime/Runtime.scala (hunk 16 above)
Comment (JaroslavTulach) [reply engine/polyglot-api/src/main/scala/org/enso/polyglot/runtime/Runtime.scala:623]: Documented in [1042ac9](https://github.com/enso-org/enso/pull/7115/commits/1042ac9e24f9d4c58f8f40a7f5dbbce698b4361e)

File: app/gui/src/controller/graph/executed.rs [hunk 19]
```diff
@@ -405,10 +405,19 @@ impl Handle {
...
     /// module's path and the `project` name.
-    pub fn module_qualified_name(&self, project: &dyn model::project::API) -> QualifiedName {
+    pub fn module_qualified_name_with_project(
+        &self,
+        project: &dyn model::project::API,
+    ) -> QualifiedName {
         self.graph().module.path().qualified_module_name(project.qualified_name())
     }
 
+    /// Get a full qualified name of the module in the [`graph`]. The name is obtained from the
```
Comment (hubertp) [inline app/gui/src/controller/graph/executed.rs:415]:
```suggestion
    /// Get a fully qualified name of the module in the [`graph`]. The name is obtained from the
```

File: app/gui/src/controller/graph/executed.rs [hunk 20]
```diff
@@ -405,10 +405,19 @@ impl Handle {
 
     /// Get a full qualified name of the module in the [`graph`]. The name is obtained from the
```
Comment (hubertp) [inline app/gui/src/controller/graph/executed.rs]:
```suggestion
    /// Get a fully qualified name of the module in the [`graph`]. The name is obtained from the
```

File: engine/language-server/src/main/scala/org/enso/languageserver/runtime/VisualizationConfiguration.scala (hunk 18 above)
Comment (JaroslavTulach) [reply engine/language-server/src/main/scala/org/enso/languageserver/runtime/VisualizationConfiguration.scala:108]: Documentation updated in 7b271889afe

File: engine/polyglot-api/src/main/scala/org/enso/polyglot/runtime/Runtime.scala (hunk 16 above)
Comment (hubertp) [reply engine/polyglot-api/src/main/scala/org/enso/polyglot/runtime/Runtime.scala:623]: Missing param doc

**Review by jdunkerley — APPROVED** [review]

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala (hunk 15 above)
Comment (hubertp) [reply engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:232]: Same missing param

**Review by hubertp — COMMENTED** [review]
Some missing documentation (in the code). Other than that, engine part looks fine.

File: app/gui/src/controller/graph/executed.rs (hunk 19 above)
Comment (JaroslavTulach) [reply app/gui/src/controller/graph/executed.rs:415]: Integrated in [8806646](https://github.com/enso-org/enso/pull/7115/commits/8806646a2d8bad8b80a93692d979432c2aa361c3)

File: engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/GetAnnotationNode.java (hunk 3 above)
Comment (JaroslavTulach) [reply engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/GetAnnotationNode.java]: Support for `Text` removed in [883b17d](https://github.com/enso-org/enso/pull/7115/commits/883b17db89691352fbae2b3a7e78725e6e572a25)

**Comment by JaroslavTulach** [comment]
I believe I addressed all requests for comments and the PR is ready to be merged. Please approve.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso [hunk 21]
```diff
@@ -419,9 +419,9 @@ type_of value = @Builtin_Method "Meta.type_of"
 
    Arguments:
    - target: The value or type to get the attribute from.
-   - method_name: The name of the method or constructor to get the attribute for.
+   - method_name: The symbol representing method or constructor to get the attribute for.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso]: Maybe this should be called `method` or `method_symbol` after the changes?

File: distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso [hunk 22]
```diff
@@ -419,9 +419,9 @@ type_of value = @Builtin_Method "Meta.type_of"
...
    - parameter_name: The name of the parameter to get the attribute for.
-get_annotation : Any -> Text -> Text -> Any | Nothing
+get_annotation : Any -> Any -> Text -> Any | Nothing
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso:424]: Can we do that?

```suggestion
get_annotation : Any -> Unresolved_Symbol -> Text -> Any | Nothing
```

**Review by radeusgd — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso (hunk 22 above)
Comment (JaroslavTulach) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso:424]: Not really, I am afraid. I tried, but`Unresolved_Symbol` object doesn't have a type. There is only `Meta.Unresolved_Symbol`, but that's not the right type - it is a wrapper around the real unresolved symbol.

TL;DR: There is no better type to use than `Any`, I am afraid.

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala [hunk 23]
```diff
@@ -243,7 +249,7 @@ object UpsertVisualizationJob {
                 argumentExpressions
               ) =>
             val callback = ctx.executionService.prepareFunctionCall(
-              module,
+              expressionModule,
```
Comment (4e6) [inline engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:252]: If there are two modules now, `modue` - for evaluating arguments, and the `expressionModule` for evaluating the expression, you should use the second one when evaluating the text expression ^
```scala
          case Api.VisualizationExpression.Text(_, expression) =>
            val callback = ctx.executionService.evaluateExpression(
              module,
              expression
            )
```
https://github.com/enso-org/enso/pull/7115/files#diff-3dbde2d45b1e4c5cf9dabfd23c1e7d89896ccac7ee7b8b694d65c1d6fd950037R240-R244

File: distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso (hunk 22 above)
Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso:424]: I see, ok.

File: docs/language-server/protocol-language-server.md [hunk 24]
```diff
@@ -453,10 +453,9 @@ interface VisualizationConfiguration {
   executionContextId: UUID;
 
   /**
-   * A qualified name of the module containing the expression which creates
-   * visualization.
+   * A qualified name of the module containing the value to be visualized.
```
Comment (4e6) [inline docs/language-server/protocol-language-server.md]: If I understand correctly, the `visualizationModule` does not contain the value, it is used to evaluate the arguments of the visualization expression.

Comment (4e6) [reply docs/language-server/protocol-language-server.md]: Please clarify this detail in the doc

**Review by 4e6 — APPROVED** [review]
I left a couple of comments on what I believe needs to be fixed. Other than that LGTM

File: docs/language-server/protocol-language-server.md (hunk 24 above)
Comment (JaroslavTulach) [reply docs/language-server/protocol-language-server.md]: Done in eba65de7f3c749b887e90227bd23540a66bf9d48

Comment (JaroslavTulach) [reply docs/language-server/protocol-language-server.md:456]: Is this what you want: 38c486e110da8c3fd5b0b5ee81494a3e71f03f97 ?

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala (hunk 23 above)
Comment (JaroslavTulach) [reply engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:252]: Is this what you are requesting: 38c486e110da8c3fd5b0b5ee81494a3e71f03f97 ?
```diff
enso$ git show
commit 38c486e110da8c3fd5b0b5ee81494a3e71f03f97 (HEAD -> wip/jtulach/UnresolvedSymbol_6955)
Author: Jaroslav Tulach <jaroslav.tulach@enso.org>
Date:   Tue Jun 27 15:51:08 2023 +0200

    Evaluate expression in expressionModule

diff --git engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala
index 9dfabf21b8..1d87cdc93e 100644
--- engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala
+++ engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala
@@ -239,7 +239,7 @@ object UpsertVisualizationJob {
         val (callback, arguments) = expression match {
           case Api.VisualizationExpression.Text(_, expression) =>
             val callback = ctx.executionService.evaluateExpression(
-              module,
+              expressionModule,
               expression
             )
             val arguments = Vector()
```

**Review by Frizi — APPROVED** [review]
